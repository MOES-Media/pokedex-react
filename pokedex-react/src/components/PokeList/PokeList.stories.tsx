import React from 'react';
import { Meta, Story } from '@storybook/react';

import PokeList, {PokeListProps, Pokemon} from '.';

export default {
    title: "Components/PokeList",
    component: PokeList
} as Meta;

const pokemons: Pokemon[] = [
    {
        id: 1,
        sprite:
          "https://raw.githubusercontent.com/PokeAPI/sprites/master/sprites/pokemon/1.png",
      },{
        id: 2,
        sprite:
          "https://raw.githubusercontent.com/PokeAPI/sprites/master/sprites/pokemon/2.png",
      },
  ];

const Template: Story<PokeListProps> = (args) => <PokeList {...args} />;
export const Default = Template.bind({});
Default.args = {
    pokemons
};