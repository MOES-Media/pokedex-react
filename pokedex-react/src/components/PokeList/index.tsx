import React, { FC } from "react";
import styled from "styled-components";
import PokeListItem from "./../PokeListItem";

export interface Pokemon {
  id: number;
  sprite: string;
}

export interface PokeListProps {
  pokemons: Pokemon[];
}

const StyledPokeList = styled.div`
  background-color: #bff9ff;
  box-shadow: inset 0 0 20px rgba(0, 0, 0, 0.5);
  padding: 1rem;
  display: flex;
  justify-content: space-between;
`;

const handlePokeListItemClick = () => {
  console.log("you have clicked a poke list item!");
};

const PokeList: FC<PokeListProps> = ({ pokemons }) => (
  <StyledPokeList>
    {pokemons.map((pokemon: Pokemon) => (
      <PokeListItem
        sprite={pokemon.sprite}
        onClick={handlePokeListItemClick}
        key={pokemon.id}
      />
    ))}
  </StyledPokeList>
);

export default PokeList;
