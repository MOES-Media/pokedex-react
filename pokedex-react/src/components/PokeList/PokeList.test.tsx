import React from "react";
import renderer from 'react-test-renderer';

import PokeList, { Pokemon } from ".";

const mockPokemons: Pokemon[] = [
  {
    id: 1,
    sprite: "foobar",
  },
  {
    id: 2,
    sprite: "foobar",
  },
];

test("Component: PokeList should render correctly", () => {
    const snapshot = renderer.create(<PokeList pokemons={mockPokemons} />);
    expect(snapshot).toMatchSnapshot();
});