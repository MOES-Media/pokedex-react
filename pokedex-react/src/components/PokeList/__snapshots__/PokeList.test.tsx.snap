// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Component: PokeList should render correctly 1`] = `
<div
  className="sc-gtsrHT fnGONm"
>
  <button
    className="sc-bdnxRM kyGyEC"
    onClick={[Function]}
    role="button"
  />
  <button
    className="sc-bdnxRM kyGyEC"
    onClick={[Function]}
    role="button"
  />
</div>
`;
